x/stakeibc/ibc_middleware.go

```diff
@@ -90,52 +91,93 @@ func (im IBCMiddleware) OnChanOpenAck(
 
-	// addresses
-	withdrawalAddress, err := icatypes.NewControllerPortID(types.FormatICAAccountOwner(hostChainId, types.ICAAccountType_WITHDRAWAL))
+	// expected port IDs for each ICA account type
+	withdrawalPortID, err := icatypes.NewControllerPortID(types.FormatICAAccountOwner(hostChainId, types.ICAAccountType_WITHDRAWAL))
 	if err != nil {
 		return err
 	}
-	feeAddress, err := icatypes.NewControllerPortID(types.FormatICAAccountOwner(hostChainId, types.ICAAccountType_FEE))
+	feePortID, err := icatypes.NewControllerPortID(types.FormatICAAccountOwner(hostChainId, types.ICAAccountType_FEE))
 	if err != nil {
 		return err
 	}
-	delegationAddress, err := icatypes.NewControllerPortID(types.FormatICAAccountOwner(hostChainId, types.ICAAccountType_DELEGATION))
+	delegationPortID, err := icatypes.NewControllerPortID(types.FormatICAAccountOwner(hostChainId, types.ICAAccountType_DELEGATION))
 	if err != nil {
 		return err
 	}
-	redemptionAddress, err := icatypes.NewControllerPortID(types.FormatICAAccountOwner(hostChainId, types.ICAAccountType_REDEMPTION))
+	redemptionPortID, err := icatypes.NewControllerPortID(types.FormatICAAccountOwner(hostChainId, types.ICAAccountType_REDEMPTION))
 	if err != nil {
 		return err
 	}
 	communityPoolDepositOwner := types.FormatICAAccountOwner(hostChainId, types.ICAAccountType_COMMUNITY_POOL_DEPOSIT)
-	communityPoolDepositAddress, err := icatypes.NewControllerPortID(communityPoolDepositOwner)
+	communityPoolDepositPortID, err := icatypes.NewControllerPortID(communityPoolDepositOwner)
 	if err != nil {
 		return err
 	}
 	communityPoolReturnOwner := types.FormatICAAccountOwner(hostChainId, types.ICAAccountType_COMMUNITY_POOL_RETURN)
-	communityPoolReturnAddress, err := icatypes.NewControllerPortID(communityPoolReturnOwner)
+	communityPoolReturnPortID, err := icatypes.NewControllerPortID(communityPoolReturnOwner)
 	if err != nil {
 		return err
 	}
 
 	// Set ICA account addresses
 	switch {
-	case portID == withdrawalAddress:
+	case portID == withdrawalPortID:
 		zoneInfo.WithdrawalIcaAddress = address
-	case portID == feeAddress:
+	case portID == feePortID:
 		zoneInfo.FeeIcaAddress = address
-	case portID == delegationAddress:
+	case portID == delegationPortID:
 		zoneInfo.DelegationIcaAddress = address
-	case portID == redemptionAddress:
+	case portID == redemptionPortID:
 		zoneInfo.RedemptionIcaAddress = address
-	case portID == communityPoolDepositAddress:
+	case portID == communityPoolDepositPortID:
 		zoneInfo.CommunityPoolDepositIcaAddress = address
-	case portID == communityPoolReturnAddress:
+	case portID == communityPoolReturnPortID:
 		zoneInfo.CommunityPoolReturnIcaAddress = address
 	default:
 		ctx.Logger().Error(fmt.Sprintf("Missing portId: %s", portID))
 	}
 
+	// Once the delegation channel is registered, whitelist epochly transfers so they're not rate limited
+	// Epochly transfers go from the deposit address to the delegation address
+	if portID == delegationPortID {
```

Comment:
Why are you whitelisting the delegation and the fee address here?  For most human kicked off staking actions we would want the security that having a rate limit gives normally.  I assume you are whitelisting these because our call to stake and redeem in the community pool route also goes through -- is there any way we can make just those calls unlimited but keep the normal staking actions limited?

Reply:
great question. So for context, I just copied over what we did for the other host zones awhile back. And the two non-community pool related whitelists are to make sure our normal epochly flow during regular liquid stakes don't trip the rate limit. 

The fee address here is whitelisted here because stToken rewards to STRD stakers go through that route. And then the delegation address is whitelisted cause that's where the epochly liquid stakes flow through. 

The part I can't quite remember is why we have the deposit → delegation whitelist when we only whitelist stTokens at the moment. But I think it may have just been precautionary in case we whitelisted native tokens in the future